[Add]test of task system

Megumi-N opened the pull request:
Rspec編課題4についてpull requestします。
ご確認のほどよろしくお願いします。

Megumi-N committed:
[Add]test of task system

yuji91 commented on spec/factories/tasks.rb:
```diff
@@ -6,4 +6,10 @@
     deadline { Random.rand(from..to) }
   end
+
+  trait :task_finished_status do
```
「ステータスが完了になっていること」以外にも「完了日時が登録されていること」などの要素があるので「finished」などのtrait名の方が分かりやすいかもしれません。

また今回は`done`というステータスがあるので、`finished`よりはステータス名の`done`に合わせた方が良いでしょう。
```suggestion
  trait :done do
```

yuji91 commented on spec/factories/tasks.rb:
```diff
@@ -6,4 +6,10 @@
+  trait :task_finished_status do
+    status { :done }
+    completion_date { Time.current.yesterday }
+    association :project
```
associationはこのtraitに限定せず、このFactory全体に適用させて良いと思います。

yuji91 commented on spec/system/task_spec.rb:
```diff
@@ -1,27 +1,33 @@
 
 RSpec.describe 'Task', type: :system do
+  let(:project) { create(:project) }
+  let(:task) { create(:task, project: project) }
```
Factoryの指摘事項を対応することで省略可能です。
```suggestion
  let(:task) { create(:task) }
```

yuji91 commented on spec/system/task_spec.rb:
```diff
@@ -59,22 +62,19 @@
         click_link 'Back'
-        expect(find('.task_list')).to have_content(Time.current.strftime('%Y-%m-%d'))
+        # 空うめは_を使う https://docs.ruby-lang.org/ja/latest/class/Time.html#S_LOCAL
+        expect(find('.task_list')).to have_content(Time.current.strftime('%_m/%d %H:%M'))
```
NG例のPRを確認して欲しいのですが、この部分を変更するとView側の表示メソッドの仕様変更時にテストが落ちてしまいます。

yuji91 commented on spec/system/task_spec.rb:
```diff
@@ -100,13 +100,12 @@
+        # click_link 'Destroy'
+        # page.driver.browser.switch_to.alert.accept #alartをaccept
+        accept_alert { click_link 'Destroy' } #alartをaccept
+        expect(find '.task_list').not_to have_content task.title #.task_listの中にtask.titleが入ってないことを確認。以下のexpectもこの指定が適応される
```
修正方法は問題ないですが「以下のexpectもこの指定が適応される」という点はどの事象から判断された形になりますでしょうか？

Megumi-N replied:
そのように考えたのですが、質問コーナーで認識違いということがわかりました。
上記のコメントアウトの記載は削除します。

yuji91 reviewed (changes requested):
修正はほとんど問題なかったですが、いくつか指摘しました！

Megumi-N committed:
[Fix]change trait :task_finished_status -> trait :done
applied association for whole factory task
delete project: project of let(:task) { create(:task, project: project) }
fix from Time.current.strftime() to short_time(), task編集-正常系-taskを編集した場合一覧画面で編集後の内容が表示

yuji91 reviewed (approved):
修正確認しました。

指摘事項があった際は、指摘内容ごとに修正のコミットを分けられると、確認がしやすくて良いと思います！
